force-app/main/default/classes/InterviewEventTriggerTest.cls

```diff
@@ -1,12 +1,65 @@
 @isTest
 public class InterviewEventTriggerTest {
-    @isTest static void InterviewEventTriggerInsert() {
 
-        Interview__c interview1 = new Interview__c(Name='Movie 1', InterviewStartDate__c= date.today(), Duration__c= 1);
+    @TestSetup
+    static void setup(){
+         TestDataFactory.defaultValueProvider = new MyDefaultValueProvider();
 
-        insert interview1;
+        Candidate__c newCanditate = (Candidate__c)TestDataFactory.createSObject('Candidate__c', new Map<String,Object>{
+        'First_Name__c' => 'Test',
+        'Last_Name__c' => 'Candidate'
+        });
 
-        System.debug(date.today());   
+        List<Interview__c> testInterviewsList = (List<Interview__c>)TestDataFactory.createSObjectList('Interview__c', new Map<String,Object>{
+            'Name' =>  'Interview {!index}',
+            // 'Candidate__cId' => TestDataFactory.DEFAULT_VALUE,
+            'Duration__c' => 2,
+            'InterviewStartDateTime__c' => TestDataFactory.DEFAULT_VALUE
+          },200);
+    }
+
+    @isTest static void CheckEventEndDateTimeForInterviewAfterInsertTest() {
 
+        Interview__c interview = [Select Id, Name__c, InterviewStartDateTime__c, Duration__c, InterviewEndDateTime__c From Interview__c where Name = 'Interview 1'];
```

Comment:
The remaining topics:
***Test as well should be bulkified not only data preparation
***Use of a test user (A user that you create in test setup and you control its profile, role...) which simulates the real context (for example a sales user which should not have system admin but another profile (Sales User profile for example). Then you wrap your test with System.runAs (https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_tools_runas.htm)
***Wrap your test scenario (just the insert part or update) with Test.startTest() and Test.stopTest() to have governor limits dedicated to your test (https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_tools_start_stop_test.htm)